packages/flutter/lib/src/widgets/image.dart

```diff
@@ -110,6 +113,10 @@ class Image extends StatefulWidget {
   ///
   /// The [image], [alignment], [repeat], and [matchTextDirection] arguments
   /// must not be null.
+  ///
+  /// The [width] and [height] arguments should be specified. If they are not,
```

Comment:
I think the advice we want to provide is: don't allow the layout to be defined by the image's actual size. Either constrain the size of the image or specify it explicitly with width and height.

Reply:
How about this:
```
  /// Either the [width] and [height] arguments should be specified, or the                                                             
  /// widget should be placed in a context that sets tight layout constraints.                                                          
  /// Otherwise, the image dimensions will change as the image is loaded, which                                                         
  /// will result in ugly layout changes.                                                                                               ```